packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -0,0 +1,239 @@
+import iam = require('@aws-cdk/aws-iam');
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { IDatabase } from './database';
+import { CfnTable } from './glue.generated';
+import { Column } from './schema';
+import { StorageType } from './storage-type';
+
+export interface ITable extends cdk.IConstruct {
+  readonly tableArn: string;
+  readonly tableName: string;
+
+  export(): TableImportProps;
+}
+
+export interface TableImportProps {
+  tableArn: string;
+  tableName: string;
+}
+
+export interface TableProps {
+  /**
+   * Name of the table.
+   */
+  tableName: string;
+
+  /**
+   * Description of the table.
+   *
+   * @default generated
+   */
+  description?: string;
+
+  /**
+   * Database in which to store the table.
+   */
+  database: IDatabase;
+
+  /**
+   * S3 bucket in which to store data.
+   *
+   * @default one is created for you
+   */
+  bucket?: s3.Bucket;
+
+  /**
+   * S3 prefix under which table objects are stored.
+   *
+   * @default data/
+   */
+  prefix?: string;
+
+  /**
+   * Columns of the table.
+   */
+  columns: Column[];
+
+  /**
+   * Partition columns of the table.
+   *
+   * @default table is not partitioned
+   */
+  partitionKeys?: Column[]
+
+  /**
+   * Storage type of the table's data.
+   */
+  storageType: StorageType;
+
+  /**
+   * Indicates whether the table's data is compressed or not.
+   *
+   * @default false
+   */
+  compressed?: boolean;
+
+  /**
+   * Indicates whether the table data is stored in subdirectories.
+   *
+   * @default false
+   */
+  storedAsSubDirectories?: boolean;
+}
```

Comment:
What about encryption? Is that implicit through the bucket?

Reply:
It's implicit through the bucket except for Client-Side Encryption (CSE): https://docs.aws.amazon.com/athena/latest/ug/encryption.html

Taking a second look, I think it makes sense to declare a new `enum TableEncryption` specific to Glue terms:
* `SSE-S3` - server-side encryption with key managed by S3 (`S3Managed`)
* `SSE-KMS` - server-side encryption with a key managed by KMS (`KmsManaged`/`Kms`)
* `CSE-KMS` - client-side encryption (no corresponding option on `Bucket`

I'm wondering why `s3.Bucket` distinguishes between a key managed by KMS vs managed by the customer (`KmsManaged` and `Kms`) when it can be distinguished with `encryptionKey === undefined` ...

Reply:
Oh I see, it's so we know when to automatically create a KMS Key,